paddle/fluid/imperative/layer.cc

```diff
@@ -132,8 +132,8 @@ class Autograd {
       return;
     }
     VLOG(3) << "start autograd";
-    bck_map = new BackwardSumMap();
-    grad_ref = new GradientRef();
+    bck_map = BackwardSumMap();
+    grad_ref = GradientRef();
```

Comment:
These two lines are unnecessary. BTW, it seems that it is not necessary that`bck_map` and `grad_ref` are data member of `Autograd`. Just make them temporary variable inside the function.